### ISSUE TITLE
All AnonAddy domains blocked by Apple

### ISSUE THREAD

**Issue by pjllorens** [anonaddy/anonaddy#477]
I'm a subscribed user. As the title says, tried to change my email on an Apple account, checked with all options of domains (shared and custom) and are not recognized as a valid email by Apple.

**Comment by MrPeteypoo** [comment]
I had the exact same issue with a custom domain

**Comment by Torkurson** [comment]
Maybe Apple is checking the CNAME or MX records on domains? I hope this practice does not become more widespread.

**Comment by businessBoris** [comment]
Excuse the interruption: am I correct in thinking that the issue is more widespread than Apple incoming?
Apple has 'blacklisted' anonaddy domains (custom and specific) for outgoing email sent by their customers?

@Torkurson 
Before taking up AnonAddy, I used something similar and much less flexible but proprietry: IronVest (aka abine, do not track, and others). They haven't had to change their remailing domain name for a couple of years. Before that they used quite a few different proxy domains.

I don't recall ever having problems writing to Apple domains with it, although the account name part of the email address, before the @ sign, is very obviously computer generated ie. not directly a real person.
I used/use Abine for a decade since 2013; I haven't transferred all the email addresses over to AnonAddy yet! It's too exciting for me to do too many at once.

**Comment by Torkurson** [comment]
Same issue with Jetbrains, even with custom domains.

**Comment by grandeljay** [comment]
I had this issue with Discord. I think their E-Mail input regex (pattern for recognising the email format) was just bad:

* discord@foo.anonaddy.me did not work
* discord@custom.domain did work

Maybe it's the same with Apple? Either way, I'd try contacting them and telling them about your E-Mail (format) not being recognised.

**Comment by rokejulianlockhart** [comment]
Maybe AnonAddy will have to end up just hosting their own open-to-use TLD and giving us all domains from that? Surely Apple wouldn't block a TLD being used for multiple things, regardless of the owner?

**Comment by mamciek** [comment]
I think this is related: https://github.com/anonaddy/anonaddy/issues/583

**Comment by RokeJulianLockhart** [comment]
> https://github.com/anonaddy/anonaddy/issues/477#issuecomment-1752091878

@QJ I don't think that that's relevant to *this* issue, although you're correct. See https://github.com/mastodon/mastodon/issues/27436#issue-1945559401.

**Comment by ComicMango** [comment]
Trying to change my Apple account email to a custom domain alias to no avail; other custom domains not on Addy worked.

**Comment by beedell-roke** [comment]
> [`#issuecomment-1752091878`](https://github.com/anonaddy/anonaddy/issues/477#issuecomment-1752091878)

I've ascertained a workaround: Use [Firefox Relay](https://relay.firefox.com/accounts/profile/?) for your Apple account with an Addy address set as your the e-mail address for your Firefox account. Firefox Relay isn't blocked, and successfully routes all mail. This functions for [`icloud.com/settings`](https://www.icloud.com/settings/) and [`mastodon.social/auth/edit`](https://mastodon.social/auth/edit#:~:text=Security-,E%2Dmail%20address,-*). Though, note that this doesn't work for what [`meta.discourse.org/400900/2`](https://meta.discourse.org/t/i-am-unable-to-register-for-a-free-discourse-site/400900/2?u=rokejulianlockhart) discusses.

However, [this doesn't work everywhere](https://github.com/mozilla/fx-private-relay/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22website%20compat%20bug%22).

**Comment by noughtmare** [comment]
GitHub and Steam also block anonaddy.io.

**Comment by noughtmare** [comment]
@RokeJulianLockhart lucky for you I guess...

**Comment by RokeJulianLockhart** [comment]
> https://github.com/anonaddy/anonaddy/issues/477#issuecomment-2717302733

@noughtmare, have you tried `addy.io` instead? That's what I utilised.

**Comment by noughtmare** [comment]
@RokeJulianLockhart thank you! I was using the free plan, so I didn't see that option. Now, I'm on Lite and using addy.io indeed works for GitHub at least, will try Steam next. Perhaps this should be advertised more.

**Comment by noughtmare** [comment]
https://tweakers.net also blocks alias addresses (even the @addy.io addresses)

**Comment by RokeJulianLockhart** [comment]
Tangentially, as [`issues/844`](https://github.com/anonaddy/anonaddy/issues/844#issue-4337611185) explains, if anyone here has created a European Commission WebGate ("EU Login") account with an Addy e-mail address, immediately replace it whilst you retain access. Ideally, create a support ticket before you do, in case you become locked out. Otherwise, you'll be unable to access it thereafter, without recourse.

**Comment by sebastianpopp** [comment]
I'm wondering if self hosted instances face the same issues. In case of a custom domain, services like verifymail.io can easily detect that anonaddy is behind it by denylisting the mx records hostnames and/or ip addresses.

Maybe a solution could be to allow running a proxy like HAProxy for the smtp service on a cheap vps with it's own ip address, so the mx records are not affiliated with anonaddy anymore. For incoming mail that might even be possible without any changes to the anonaddy codebase. For outgoing mail (especially with DKIM) a deeper integration would be needed.
